containers/Makefile

```diff
@@ -59,42 +64,30 @@ $(info IMAGE_PREFIX=$(IMAGE_PREFIX))
 
 all: nginx-proxy nginx-storage concent-api postgresql verifier
 
-nginx-proxy: build/ $(NGINX_PROXY_SOURCE) build/nginx-proxy/concent-api-assets.tar build/nginx-proxy/concent-api-docs.tar
-	docker build --tag $(IMAGE_PREFIX)nginx-proxy:$(IMAGE_VERSION) build/nginx-proxy/
-	docker tag $(IMAGE_PREFIX)nginx-proxy:$(IMAGE_VERSION) $(IMAGE_PREFIX)nginx-proxy:latest
-
-nginx-storage: build/ $(NGINX_STORAGE_SOURCE)
-	docker build --tag $(IMAGE_PREFIX)nginx-storage:$(IMAGE_VERSION) build/nginx-storage/
-	docker tag $(IMAGE_PREFIX)nginx-storage:$(IMAGE_VERSION) $(IMAGE_PREFIX)nginx-storage:latest
-
-postgresql: build/ $(POSTGRESQL_SOURCE)
-	docker build --tag $(IMAGE_PREFIX)postgresql:$(IMAGE_VERSION) build/postgresql/
-	docker tag $(IMAGE_PREFIX)postgresql:$(IMAGE_VERSION) $(IMAGE_PREFIX)postgresql:latest
-
-concent-api: build/ $(CONCENT_API_SOURCE) build/concent-api/requirements.lock build/concent-api/testing.pref build/repositories/concent/
-	docker build --tag $(IMAGE_PREFIX)concent-api:$(IMAGE_VERSION) build/concent-api/
-	docker tag $(IMAGE_PREFIX)concent-api:$(IMAGE_VERSION) $(IMAGE_PREFIX)concent-api:latest
-
-verifier: build/ $(VERIFIER_SOURCE) build/repositories/golem/ build/verifier/golem/scripts/ build/verifier/golem/imgverifier-requirements.txt
-	docker build --tag $(IMAGE_PREFIX)verifier:$(IMAGE_VERSION) build/verifier/
-	docker tag $(IMAGE_PREFIX)verifier:$(IMAGE_VERSION) $(IMAGE_PREFIX)verifier:latest
+nginx-proxy: $(NGINX_PROXY_SOURCE) build/nginx-proxy/concent-api-assets.tar build/nginx-proxy/concent-api-docs.tar | build/
+	$(DOCKER_BUILD)
+	$(DOCKER_TAG)
 
-docker-clean: nginx-proxy-clean concent-api-clean postgresql-clean verifier-clean
+nginx-storage: $(NGINX_STORAGE_SOURCE) | build/
+	$(DOCKER_BUILD)
+	$(DOCKER_TAG)
```

Comment:
This does reduce duplication a bit but has two downsides:
- You still have to repeat those three lines for every image.
- The variables used like this look quite strange and opaque. This works but putting whole bash lines into variables feels kinda like you're trying really hard to create a function. I'd rather create a helper bash script and call it than do it this way.

I was actually thinking about doing it like we did in one of our earlier projects. First define a list of target names and then automatically declare the targets with a [Static Pattern Rule](https://www.gnu.org/software/make/manual/html_node/Static-Pattern.html#Static-Pattern). Prerequisites can be added using a special intermediate target.

Here's an incomplete example:

``` make
IMAGE_TARGETS := \
    nginx-proxy  \
    concent-api  \
    postgresql

nginx-proxy!dependencies:                    \
    $(NGINX_PROXY_SOURCE)                    \
    build/nginx-proxy/concent-api-assets.tar \
    build/nginx-proxy/concent-api-docs.tar

concent-api!dependencies:               \
    $(CONCENT_API_SOURCE)               \
    build/concent-api/requirements.lock \
    build/concent-api/testing.pref      \
    build/repositories/concent/

postgresql!dependencies: \
    $(POSTGRESQL_SOURCE)

IMAGE_DEPENDENCY_TARGETS := $(addsuffix !dependencies, $(IMAGE_TARGETS))

all: images

$(IMAGE_TARGETS): %: %!dependencies build/
    docker build --tag $(IMAGE_PREFIX)$@:$(IMAGE_VERSION) build/$@/
    docker tag $(IMAGE_PREFIX)$@:$(IMAGE_VERSION) $(IMAGE_PREFIX)$@:latest

.PHONY:                         \
    all                         \
    $(IMAGE_TARGETS)            \
    $(IMAGE_DEPENDENCY_TARGETS)
```